File: docs/guides/java/src/test/java/com/algolia/compression.java
package com.algolia;

import com.algolia.api.SearchClient;
import com.algolia.config.*;
import com.algolia.model.search.*;

public class compression {

  public static void main(String[] args) throws Exception {
    // Initialize the client with gzip compression enabled
    // Compression reduces the size of request bodies sent to Algolia
    SearchClient client = new SearchClient(
      "ALGOLIA_APPLICATION_ID",
      "ALGOLIA_API_KEY",
      ClientOptions.builder().setCompressionType(CompressionType.GZIP).build()
    );

    // Search with compressed request body
    var result = client.searchSingleIndex("<YOUR_INDEX_NAME>", new SearchParamsObject().setQuery("comedy"), Hit.class);
    System.out.println(result.getHits());
    client.close();
  }
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/compression.kt
package org.example

import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

suspend fun main() {
  // Initialize the client with gzip compression enabled
  // Compression reduces the size of request bodies sent to Algolia
  val client =
    SearchClient(
      appId = "ALGOLIA_APPLICATION_ID",
      apiKey = "ALGOLIA_API_KEY",
      options = ClientOptions(compressionType = CompressionType.GZIP),
    )

  // Search with compressed request body
  try {
    val result =
      client.searchSingleIndex(
        indexName = "<YOUR_INDEX_NAME>",
        searchParams = SearchParamsObject(query = "comedy"),
      )
    println(result.hits)
  } catch (e: Exception) {
    println(e.message)
  }
}
